nexus/src/app/sagas/switch_port_settings_common.rs

```diff
@@ -454,11 +456,44 @@ pub(crate) async fn ensure_switch_port_bgp_settings(
                 ActionError::action_failed(format!("get bgp config: {e}"))
             })?;
 
+        if let Some(cfg) = &cfg {
+            if config.asn != cfg.asn {
+                return Err(ActionError::action_failed(
+                    "bad request: only one AS allowed per switch".to_string(),
+                ));
+            }
+        } else {
+            cfg = Some(config);
+        }
+
+        let bpc = BgpPeerConfig {
+            name: format!("{}", peer.addr.ip()), //TODO user defined name?
+            host: format!("{}:179", peer.addr.ip()),
+            hold_time: peer.hold_time.0.into(),
+            idle_hold_time: peer.idle_hold_time.0.into(),
+            delay_open: peer.delay_open.0.into(),
+            connect_retry: peer.connect_retry.0.into(),
+            keepalive: peer.keepalive.0.into(),
+            resolution: BGP_SESSION_RESOLUTION,
+            passive: false,
+        };
+
+        match bgp_peer_configs.get_mut(&switch_port_name) {
```

Comment:
You could use `HashMap::entry` with `Entry::or_insert_with` to be a bit more succint, but I suppose you'd need to clone `switch_port_name` unconditionally.

Reply:
Yeah I think this is the equivalent using the `Entity::or_*` API

```rust
bgp_peer_configs.entry(switch_port_name.clone()).or_default().push(bpc);
```

And while that is more concise, I would say it's bordering on terse/dense, and it's less clear what the intent is. It's also very possible that I'm too dense to understand how to use this API in a clear way 😆.